pkg/test/framework/components/istio/ingress.go

```diff
@@ -49,22 +49,115 @@ var (
 	retryTimeout = retry.Timeout(3 * time.Minute)
 	retryDelay   = retry.Delay(5 * time.Second)
 
-	_ Instance = &kubeComponent{}
+	_ Ingress = &ingress{}
 )
 
-type kubeComponent struct {
-	id        resource.ID
+// CallType defines ingress gateway type
+type CallType int
+
+const (
+	PlainText CallType = iota
+	TLS
+	Mtls
+)
+
+// CallOptions defines options for calling a Endpoint.
+type CallOptions struct {
```

Comment:
nit: `IngressCallOptions`? Alternatively, we could still define the API in the `ingress` package and then have the implementation here.

Reply:
I ended up moving it.